### ISSUE TITLE
Panic in binaryscalarexpr.go with array.Dictionary

### ISSUE THREAD

**Issue by metalmatze** [polarsignals/frostdb#322]
There seems to be a panic when querying with the dictionary types:
https://github.com/polarsignals/frostdb/blob/395932c312ca14c228bc94292fba8930efaede3c/query/physicalplan/binaryscalarexpr.go#L144

It seems like we should support more of the logicalplan operators. Sadly it's not obvious from the stacktrace at which one we should the closet. 

```
panic: unsupported value comparison: Type(?) [recovered]
	panic: unsupported value comparison: Type(?) [recovered]
	panic: unsupported value comparison: Type(?)


To see all goroutines, visit https://github.com/maruel/panicparse#gotraceback

1: running [Created by errgroup.(*Group).Go @ golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72]
    trace    go.opentelemetry.io/otel/sdk@v1.11.2/trace/span.go:383                                     (*recordingSpan).End.func1()
    trace    go.opentelemetry.io/otel/sdk@v1.11.2/trace/span.go:421                                     (*recordingSpan).End(0xc0524d8f00, {0, 0, 0x1d6cbdd500a4d2d8})
             runtime/panic.go:884                                                                       panic({#1, #7})
    trace    go.opentelemetry.io/otel/sdk@v1.11.2/trace/span.go:383                                     (*recordingSpan).End.func1()
    trace    go.opentelemetry.io/otel/sdk@v1.11.2/trace/span.go:421                                     (*recordingSpan).End(0xc056039500, {0, 0, 0})
             runtime/panic.go:884                                                                       panic({#1, #7})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/binaryscalarexpr.go:144 compare({0, 0, 0, 0, 0, 0}, {0xc0974ed930, 0x10, 0xf9, 0, ...})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/binaryscalarexpr.go:120 Max({#4, #6})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/binaryscalarexpr.go:75  BinaryScalarOperation({#4, #6}, {#8, 0x13, 0xf9, 0, 0, 0}, 0x19e8728)
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/binaryscalarexpr.go:64  BinaryScalarExpr.Eval({0xc0524add20, 1, {#8, 0x13, 0xf9, 0, 0, 0}}, {#10, #9})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:132           (*AndExpr).Eval(0xc0524c5a60, {#10, #9})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140           (*AndExpr).Eval(0xc0524c5a80, {#10, #9})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140           (*AndExpr).Eval(0xc0524c5aa0, {#10, #9})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140           (*AndExpr).Eval(0xc0524c5ac0, {#10, #9})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140           (*AndExpr).Eval(0xc0524c5ae0, {#10, #9})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140           (*AndExpr).Eval(0xc0524c5b00, {#10, #9})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140           (*AndExpr).Eval(0xc0524c5b20, {#10, #9})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140           (*AndExpr).Eval(0xc0524c5b40, {#10, #9})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140           (*AndExpr).Eval(#5, {#10, #9})
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/store.go:141            (*Table).filterRowGroups(0xc03a254b40, {#3, 0xc062734810}, 0xc0601f2450, {#2, #5}, 0xffffffffffffffff)
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/store.go:131            (*Table).ProcessFile(0xc00112c1e0, {#3, 0xc062734210}, {0xc090023acf, 0x27}, 0x18627081fd5, {#2, #5}, 0xc0974b6058)
    frostdb  github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/store.go:63             (*Table).IterateBucketBlocks.func1.1()
    errgroup golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75                                           (*Group).Go.func1()

```

<details>
<summary>raw panic</summary>

```
panic: unsupported value comparison: Type(?) [recovered]
	panic: unsupported value comparison: Type(?) [recovered]
	panic: unsupported value comparison: Type(?)

goroutine 180205 [running]:
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End.func1()
	go.opentelemetry.io/otel/sdk@v1.11.2/trace/span.go:383 +0x2a
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End(0xc0524d8f00, {0x0, 0x0, 0x1d6cbdd500a4d2d8?})
	go.opentelemetry.io/otel/sdk@v1.11.2/trace/span.go:421 +0x942
panic({0x268aa80, 0xc06431ec30})
	runtime/panic.go:884 +0x212
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End.func1()
	go.opentelemetry.io/otel/sdk@v1.11.2/trace/span.go:383 +0x2a
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End(0xc056039500, {0x0, 0x0, 0x0?})
	go.opentelemetry.io/otel/sdk@v1.11.2/trace/span.go:421 +0x942
panic({0x268aa80, 0xc06431ec30})
	runtime/panic.go:884 +0x212
github.com/polarsignals/frostdb.compare({0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, {0xc0974ed930, 0x10, 0xf9, 0x0, ...})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/binaryscalarexpr.go:144 +0x394
github.com/polarsignals/frostdb.Max({0x3244950?, 0xc056f32c40?})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/binaryscalarexpr.go:120 +0x191
github.com/polarsignals/frostdb.BinaryScalarOperation({0x3244950, 0xc056f32c40}, {0xc068e3bcc8, 0x13, 0xf9, 0x0, 0x0, 0x0}, 0x19e8728?)
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/binaryscalarexpr.go:75 +0x1ef
github.com/polarsignals/frostdb.BinaryScalarExpr.Eval({0xc0524add20, 0x1, {0xc068e3bcc8, 0x13, 0xf9, 0x0, 0x0, 0x0}}, {0x7f6576160f18, 0xc06b1b4690})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/binaryscalarexpr.go:64 +0x1e8
github.com/polarsignals/frostdb.(*AndExpr).Eval(0xc0524c5a60, {0x7f6576160f18, 0xc06b1b4690})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:132 +0x37
github.com/polarsignals/frostdb.(*AndExpr).Eval(0xc0524c5a80, {0x7f6576160f18, 0xc06b1b4690})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140 +0x65
github.com/polarsignals/frostdb.(*AndExpr).Eval(0xc0524c5aa0, {0x7f6576160f18, 0xc06b1b4690})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140 +0x65
github.com/polarsignals/frostdb.(*AndExpr).Eval(0xc0524c5ac0, {0x7f6576160f18, 0xc06b1b4690})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140 +0x65
github.com/polarsignals/frostdb.(*AndExpr).Eval(0xc0524c5ae0, {0x7f6576160f18, 0xc06b1b4690})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140 +0x65
github.com/polarsignals/frostdb.(*AndExpr).Eval(0xc0524c5b00, {0x7f6576160f18, 0xc06b1b4690})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140 +0x65
github.com/polarsignals/frostdb.(*AndExpr).Eval(0xc0524c5b20, {0x7f6576160f18, 0xc06b1b4690})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140 +0x65
github.com/polarsignals/frostdb.(*AndExpr).Eval(0xc0524c5b40, {0x7f6576160f18, 0xc06b1b4690})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140 +0x65
github.com/polarsignals/frostdb.(*AndExpr).Eval(0xc0524c5b60, {0x7f6576160f18, 0xc06b1b4690})
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/filter.go:140 +0x65
github.com/polarsignals/frostdb.(*Table).filterRowGroups(0xc03a254b40?, {0x323a268?, 0xc062734810?}, 0xc0601f2450, {0x3214c40, 0xc0524c5b60}, 0xffffffffffffffff?)
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/store.go:141 +0x36d
github.com/polarsignals/frostdb.(*Table).ProcessFile(0xc00112c1e0, {0x323a268?, 0xc062734210?}, {0xc090023acf, 0x27}, 0x18627081fd5, {0x3214c40, 0xc0524c5b60}, 0xc0974b6058?)
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/store.go:131 +0x65f
github.com/polarsignals/frostdb.(*Table).IterateBucketBlocks.func1.1()
	github.com/polarsignals/frostdb@v0.0.0-20230127114537-7817186cf2b4/store.go:63 +0x3d
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
```

</details>

**Comment by asubiotto** [comment]
What's the latest on this? Are we still seeing it?

**Comment by asubiotto** [comment]
This was fixed by #362 I think
